duckduckgo/Android
# Fix Fireproofing on domains that authenticate elsewhere

Task/Issue URL: https://app.asana.com/0/488551667048375/1195055028680547/f
Tech Design URL: 
CC: 

**Description**:
This PR improves login detection on sites that authenticate via other domains.

We are covering the following cases:
- common social login providers
- Handling known 2FA steps on common websites
- sso via duosecurity or simpleSaml

This PR doens't enables login detection by default.

**Steps to test this PR**:

Login detection disabled:
1. Fresh install
1. Go to any website
2. Sign in
3. Ensure we don't ask to fireproof any site

Login detection enabled (social login providers):
1. Go to settings and enable login detection
2. Go to any website where you can sign in using your google/github/twitter/facebook/... account (you can take a look to some tested websites here: https://app.asana.com/0/0/1199631139904357/f)
3. Sign in 
4. Ensure we ask to fireproof the final domain

Login detection enabled (sso):
1. Go to settings and enable login detection
2. Go to Asana
3. Go to their login page (you will need to enable desktop site)
4. Sign in as usual
5. Ensure we ask to fireproof Asana domain

Login detection enabled (site form): This test is just to ensure backward compatibility, our JS layer sometimes fails detecting some login forms.
1. Go to settings and enable login detection
2. Go to any website where you have an account to sign in
4. Sign in as usual
5. Ensure we ask to fireproof that domain


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
